controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -0,0 +1,385 @@
+	cassandra "github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/secret"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Create all things Medusa related in the cassdc podTemplateSpec
+func (r *K8ssandraClusterReconciler) ReconcileMedusa(
+	ctx context.Context,
+	remoteClient client.Client,
+	dcConfig *cassandra.DatacenterConfig,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) (ctrl.Result, error) {
+	// Medusa isn't enabled (anymore) in this cluster
+	logger.Info("Medusa reconcile for " + dcConfig.Meta.Name + " on namespace " + kc.Namespace + "/" + dcConfig.Meta.Namespace)
+	medusaSpec := kc.Spec.Medusa
+	if medusaSpec != nil {
+		logger.Info("Medusa is enabled")
+		if dcConfig.PodTemplateSpec == nil {
+			dcConfig.PodTemplateSpec = &corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers:     []corev1.Container{},
+					InitContainers: []corev1.Container{},
+				},
+			}
+		}
+		if result, err := r.reconcileMedusaConfigMap(ctx, remoteClient, kc, logger); !result.IsZero() || err != nil {
+			return result, err
+		}
+		updateMedusaInitContainer(dcConfig, medusaSpec, logger)
+		updateMedusaMainContainer(dcConfig, medusaSpec, logger)
+		updateMedusaVolumes(dcConfig, medusaSpec, logger)
+	} else {
+		logger.Info("Medusa is not enabled")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// Generate a secret for Medusa or use the existing one if provided in the spec
+func (r *K8ssandraClusterReconciler) reconcileMedusaSecrets(
+	ctx context.Context,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) error {
+	logger.Info("Reconciling Medusa user secrets")
+	if kc.Spec.Medusa != nil {
+		cassandraUserSecretRef := kc.Spec.Medusa.CassandraUserSecretRef
+		if cassandraUserSecretRef == "" {
+			cassandraUserSecretRef = cassandraUserSecretName(kc.Spec.Medusa, kc.Name)
+		}
+		logger = logger.WithValues(
+			"MedusaCassandraUserSecretRef",
+			cassandraUserSecretRef,
+		)
+		if err := secret.ReconcileSecret(ctx, r.Client, cassandraUserSecretRef, kc.Name, kc.Namespace); err != nil {
+			logger.Error(err, "Failed to reconcile Medusa CQL user secret")
+			return err
+		}
+	}
+	logger.Info("Medusa user secrets successfully reconciled")
+	return nil
+}
+
+// Create the Medusa config map if it doesn't exist
+func (r *K8ssandraClusterReconciler) reconcileMedusaConfigMap(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	logger logr.Logger,
+) (ctrl.Result, error) {
+	logger.Info("Reconciling Medusa configMap on namespace : " + kc.Namespace)
+	if kc.Spec.Medusa != nil {
+		medusaIni := createMedusaIni(kc)
+		desiredConfigMap := createMedusaConfigMap(kc.Spec.Medusa, kc.Namespace, kc.Spec.Cassandra.Cluster, medusaIni)
```

Comment:
I think we should not. The configMap, just as the cassdc objects, shouldn't be modified directly otherwise it'll drift from the k8ssandracluster object definition.
Any change to the configmap should then come from modifications to the k8c object that would be reconciled.